src/components/ha-picker-combo-box.ts

```diff
@@ -256,51 +338,83 @@ export class HaPickerComboBox extends LitElement {
   private _filterChanged = (ev: Event) => {
     const textfield = ev.target as HaTextField;
     const searchString = textfield.value.trim();
+    this._search = searchString;
 
-    if (!searchString) {
-      this._items = this._allItems;
-      return;
-    }
+    if (this.sections?.length) {
+      this._items = this._getItems();
+    } else {
+      if (!searchString) {
+        this._items = this._allItems;
+        return;
+      }
+
+      const index = this._fuseIndex(this._allItems as PickerComboBoxItem[]);
+      const fuse = new HaFuse(
+        this._allItems as PickerComboBoxItem[],
+        {
+          shouldSort: false,
+          minMatchCharLength: Math.min(searchString.length, 2),
+        },
+        index
+      );
+
+      const results = fuse.multiTermsSearch(searchString);
+      let filteredItems = this._allItems;
```

Comment:
```suggestion
      let filteredItems = [ ...this._allItems ];
```
Should copy it as `filteredItems` is later mutated with `push`.